win32ss/user/user32/windows/cursoricon.c

```diff
@@ -1435,21 +1669,16 @@ CURSORICON_LoadFromFileW(
     if(!NtUserSetCursorIconData(hCurIcon, NULL, NULL, &cursorData))
     {
         NtUserDestroyCursor(hCurIcon, TRUE);
-        goto end_error;
+        hCurIcon = NULL;
+        if (cursorData.hbmMask) DeleteObject(cursorData.hbmMask);
+        if (cursorData.hbmColor) DeleteObject(cursorData.hbmColor);
+        if (cursorData.hbmAlpha) DeleteObject(cursorData.hbmAlpha);
     }
 
 end:
     UnmapViewOfFile(bits);
+    HeapFree(GetProcessHeap(), 0, pbBmpIcon);
```

Comment:
```suggestion
    HeapFree(GetProcessHeap(), 0, pbBmpIcon);
    UnmapViewOfFile(bits);
```
(should be done in reverse order of the original operations)